src/vma/sock/sockinfo_udp.cpp

```diff
@@ -563,6 +563,14 @@ int sockinfo_udp::connect(const struct sockaddr *__to, socklen_t __tolen)
 		}
 		BULLSEYE_EXCLUDE_BLOCK_END
 		on_sockname_change(name, *namelen);
+
+		if (m_rx_ring_map.size() == 1) {
+			rx_ring_map_t::iterator rx_ring_iter = m_rx_ring_map.begin();
+			m_p_rx_ring = rx_ring_iter->first;
+		} else {
+			si_udp_logdbg("ring map size: %d", m_rx_ring_map.size());
+		}
+
```

Comment:
The attach_as_uc_receiver() is uased in the on_sockname_change(name, *namelen) in this patch L.565. May be moving to the attach_as_uc_receiver() for TCP & UDP is right way.